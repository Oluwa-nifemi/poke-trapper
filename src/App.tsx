import React from 'react';
import './App.css';
import {Route, Routes} from "react-router-dom";
import Layout from "./components/layout";
import ListPokemon from "./pages/list-pokemon";
import PokemonDetails from "./pages/pokemon-details";
import MyPokemonList from "./pages/my-pokemon-list";

function App() {
    return (
        <Routes>
            <Route path="/" element={<Layout/>}>
                <Route index element={<ListPokemon/>}/>
                <Route path="/:id" element={<PokemonDetails/>}/>
                <Route path="/my-pokemon" element={<MyPokemonList/>}/>
            </Route>
        </Routes>
    );
}

export default App;
