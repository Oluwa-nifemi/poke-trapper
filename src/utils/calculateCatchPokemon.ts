const calculateCatchPokemon = (): boolean => Math.random() > 0.5

export default calculateCatchPokemon