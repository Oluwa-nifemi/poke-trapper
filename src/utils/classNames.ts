/* This function allows you to merge several classes together.
 * For conditional classes the first item in the array should be the condition (boolean)
 */
export const classNames = (...args: (string | [boolean, string])[]) => {
    return args.filter(className => {
        if (Array.isArray(className)) {
            return className[0];
        }

        return className;
    }).map((className) => {
        if (Array.isArray(className)) {
            return className[1];
        }

        return className;
    }).join(' ')
};