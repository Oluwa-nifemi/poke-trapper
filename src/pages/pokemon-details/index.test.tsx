import {render, screen} from "tests/utils";
import App from "../../App";
import {waitFor} from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import calculateCatchPokemon from "utils/calculateCatchPokemon";

const mockCalculateCatchPokemon = calculateCatchPokemon as jest.Mock;

jest.mock("utils/calculateCatchPokemon", () => {
    return jest.fn()
});

describe("Pokemon Details Test", () => {
    it('Renders the pokemon details', async function () {
        window.history.pushState({}, "", "/bulbasaur");

        const user = userEvent.setup()

        const {container} = render(<App/>);

        await waitFor(() => expect(screen.queryByText("Loading...")).toBeNull());

        expect(container).toMatchSnapshot();

        mockCalculateCatchPokemon.mockReturnValueOnce(false).mockReturnValueOnce(true)

        await user.click(screen.getByText(/catch/i))

        expect(mockCalculateCatchPokemon).toHaveBeenCalledTimes(1)

        expect(screen.getByTitle("bulbasaur was not caught : (")).toBeInTheDocument()

        await user.click(screen.getByText(/catch/i))

        expect(mockCalculateCatchPokemon).toHaveBeenCalledTimes(2)

        expect(screen.getByTitle("bulbasaur was caught!")).toBeInTheDocument()
    });
})