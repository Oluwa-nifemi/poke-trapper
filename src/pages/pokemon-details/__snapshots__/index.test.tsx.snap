// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pokemon Details Test Renders the pokemon details 1`] = `
<div>
  <nav
    class="nav"
  >
    <a
      href="/"
    >
      <svg
        class="navLogo"
      >
        logo.svg
      </svg>
    </a>
    <a
      class="navLink"
      href="/my-pokemon"
    >
      My Pokemon
    </a>
  </nav>
  <main
    class="main"
  >
    <div
      class="popup"
      role="alert"
    />
    <h1
      class="pokemonName"
    >
      bulbasaur
    </h1>
    <img
      alt="bulbasaur"
      class="pokemonImage"
      src="https://raw.githubusercontent.com/PokeAPI/sprites/master/sprites/pokemon/1.png"
    />
    <button
      class="pixelatedButton pokemonCatchButton"
    >
      <span
        class="pixelatedButtonText"
      >
        CATCH
      </span>
    </button>
    <div
      class="statsGrid"
    >
      <article
        class="statsGridItem statsGridItemBorderRight"
      >
        <h2
          class="statsGridItemName"
        >
          HEIGHT
        </h2>
        <p
          class="statsGridItemValue"
        >
          7
        </p>
      </article>
      <article
        class="statsGridItem statsGridItemBorderLeft"
      >
        <h2
          class="statsGridItemName"
        >
          WEIGHT
        </h2>
        <p
          class="statsGridItemValue"
        >
          69
        </p>
      </article>
      <article
        class="statsGridItem statsGridItemBottom"
      >
        <h2
          class="statsGridItemName"
        >
          XP
        </h2>
        <p
          class="statsGridItemValue"
        >
          64
        </p>
      </article>
    </div>
    <article
      class="attributesSection"
    >
      <h3
        class="attributesSectionHeader"
      >
        Types
      </h3>
      <ul
        class="attributesSectionList"
      >
        <li
          class="attributesSectionListItem"
        >
          grass
        </li>
        <li
          class="attributesSectionListItem"
        >
          poison
        </li>
      </ul>
    </article>
    <article
      class="attributesSection"
    >
      <h3
        class="attributesSectionHeader"
      >
        Moves
      </h3>
      <ul
        class="attributesSectionList"
      >
        <li
          class="attributesSectionListItem"
        >
          razor-wind
        </li>
        <li
          class="attributesSectionListItem"
        >
          swords-dance
        </li>
      </ul>
    </article>
  </main>
</div>
`;
